core/src/mindustry/logic/LExecutor.java

```diff
@@ -157,18 +157,29 @@ public boolean bool(int index){
         return v.isobj ? v.objval != null : Math.abs(v.numval) >= 0.00001;
     }
 
+    @Nullable
+    public double num(int index, boolean nullToNan){
```

Comment:
To avoid slowing down the common case (not nullToNan), I think this should be a separate method (e.g. `numOrNan`)